src/MoBi.Core/Services/BuildingBlockVersionUpdater.cs

```diff
@@ -51,9 +48,15 @@ private void showModuleConversionWarning(IBuildingBlock buildingBlock)
          _dialogCreator.MessageBoxInfo(AppConstants.Captions.TheModuleWillBeConvertedFromPKSimToExtensionModule(buildingBlock.Module.Name));
       }
 
-      private bool shouldShowModuleConversionWarning(IBuildingBlock buildingBlock)
+      private bool shouldShowModuleConversionWarning(IBuildingBlock buildingBlock, bool shouldConvertPKSimModule)
       {
-         return buildingBlock.Module != null && buildingBlock.Module.IsPKSimModule;
+         if (buildingBlock.Module?.IsPKSimModule == true && shouldConvertPKSimModule == true)
+         {
+            buildingBlock.Module.IsPKSimModule = false;
```

Comment:
I would not do this quite this way, the semantics of the name `shouldShowModuleConversionWarning` does not convey that anything will be changed as a result of calling the method. Instead, I would move this up one level in the call stack so you'd have

```
         if (shouldShowModuleConversionWarning(buildingBlock, shouldConvertPKSimModule))
         {
            showModuleConversionWarning(buildingBlock);
            buildingBlock.Module.IsPKSimModule = false;
         }
```

It probably makes sense then to rename the method too

```

I think that would make it easier to read.
shouldShowModuleConversionWarning -> shouldConvertModule
```

Then it reads

```
         if (shouldConvertModule(buildingBlock, shouldConvertPKSimModule))
         {
            showModuleConversionWarning(buildingBlock);
            buildingBlock.Module.IsPKSimModule = false;
         }
```